frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx

```diff
@@ -120,6 +113,12 @@ export function PropertyValue({
         loadPropertyValues('')
     }, [propertyKey])
 
+    useEffect(() => {
+        if (input === '' && document.activeElement instanceof HTMLElement) {
```

Comment:
This whole `useEffect` I'm not sure about, if you just want to erase everything to type something else, losing focus on the input seems annoying. Curious on the context for this.

Reply:
The context is a little complex so bare with me, would love any suggestions for a better way to do this:

1) Since I'm now handling the blur event in a different function, the previous code (when your press Esc key) would trigger a blur - invoking the new blur logic - which stipulates that we should persist the current value - this is not what the user would expect

2) Since the handleBlur function does not persist the value if the input is empty (which it should be after you've pressed escape), I would like to set the input value to empty, then trigger the handleBlur function, such that on escape the blur occurs after the input value has been set to empty and the value is therefore not persisted (expected behavior)

3) However, setState takes time to propagate the value and blur does not. Ideally I would've used a callback on the `setInput('')` call, however since we're using `useState`, callbacks are not supported (I think they actually work, but they're not supported officially). So I can't see an elegant way to execute the blur once the `setInput('')` call has updated the state successfully.

4) I tried another hack with a `setTimeout` but it felt mega-ugly

5) Yes, the problem is that if someone backspaces the box empty - we'll lose focus. I don't think this is a major issue - it's smaller than the existing issue this PR is designed to fix - but I would rather it didn't happen if you can think of another nice solution?

6) The only other option I considered was to refactor `setInput()` to use `setState()` rather than `useState()` and add a callback on the setState to trigger the blur - this felt like it would add unnecessary complexity - but might be the best solution

Reply:
It's great that you considered all these other options. I agree w/ Paolo that it could be annoying on input blur everytime input is empty and I think we can kill both birds w/ one stone here. If I'm understanding correctly, are you trying to blur the text field only when a user presses the escape key (and not when input becomes empty via backspace)? If that's the case we can react to a separate `shouldBlur` state. Ugly but still a lot less overhead than most other options.

Reply:
Demo

https://user-images.githubusercontent.com/13460330/127895740-39d88a79-72ef-49c7-968d-e7ca6196cc10.mov

Pushed a few changes to implement this.

Reply:
Amazing thanks @alexkim205  really appreciate both your feedback and changes.

@paolodamico since @alexkim205  has made changes here since you originally approved - are you okay with us to merge this latest version?